### PR TITLE
Do not allow different CPP dialects in CTU

### PR DESCRIPTION
**Pull request by martong**

### CODE REVIEW

File: lib/CrossTU/CrossTranslationUnit.cpp [hunk 1]
```diff
@@ -240,12 +240,34 @@ CrossTranslationUnitContext::getCrossTUDefinition(const FunctionDecl *FD,
   }
   const auto& LangTo = Context.getLangOpts();
   const auto& LangFrom = Unit->getASTContext().getLangOpts();
```
Comment (balazske) [inline lib/CrossTU/CrossTranslationUnit.cpp:242]: (I do not like `auto` here and the & is at wrong place (but not part of the change).)

**Review by balazske — APPROVED** [review]

File: lib/CrossTU/CrossTranslationUnit.cpp (hunk 1 above)
Comment (martong) [reply lib/CrossTU/CrossTranslationUnit.cpp:242]: Yes I agree, but once we will port back https://reviews.llvm.org/D55134 where this part will be updated.

**Comment by martong** [comment]
https://reviews.llvm.org/D57906
